src/modules/hip/kernel/resample.hpp

```diff
@@ -0,0 +1,171 @@
+#include <hip/hip_runtime.h>
+#include "rpp_hip_common.hpp"
+
+// -------------------- Set 0 - resample kernel --------------------
+// Single channel resample support
+__global__ void resample_1channel_hip_tensor(float *srcPtr,
+                                             float *dstPtr,
+                                             int srcLength,
+                                             int outEnd,
+                                             double scale,
+                                             RpptResamplingWindow &window,
+                                             int block)
+{
+    int id_x = hipBlockIdx_x * hipBlockDim_x + hipThreadIdx_x;
+    extern __shared__ float windowCoeffs_smem[];
+    float *windowCoeffsSh = windowCoeffs_smem;
+    for (int k = hipThreadIdx_x; k < window.lookupSize; k += block)
+        windowCoeffsSh[k] = window.lookup[k];
+    __syncthreads();
+    window.lookup = windowCoeffsSh;
+
+    int outBlock = id_x * block;
+    if (outBlock >= outEnd)
+        return;
+
+    int blockEnd = std::min(outBlock + block, outEnd);
+    double inBlockRaw = outBlock * scale;
+    int inBlockRounded = static_cast<int>(inBlockRaw);
+    float inPos = inBlockRaw - inBlockRounded;
+    float fscale = scale;
+
+    for (int outPos = outBlock; outPos < blockEnd; outPos++, inPos += fscale)
+    {
+        int loc0, loc1;
+        window.input_range(inPos, &loc0, &loc1);
+
+        if (loc0 + inBlockRounded < 0)
+            loc0 = -inBlockRounded;
+        if (loc1 + inBlockRounded > srcLength)
+            loc1 = srcLength - inBlockRounded;
+        int locInWindow = loc0;
+        float locBegin = locInWindow - inPos;
+        float accum = 0.0f;
+
+        for (; locInWindow < loc1; locInWindow++, locBegin++)
+        {
+            float w = window(locBegin);
+            accum += srcPtr[inBlockRounded + locInWindow] * w;
+        }
+        dstPtr[outPos] = accum;
+    }
+}
+
+// Generic n channel resample support
+__global__ void resample_nchannel_hip_tensor(float *srcPtr,
+                                             float *dstPtr,
+                                             int2 srcDims,
+                                             int outEnd,
+                                             double scale,
+                                             RpptResamplingWindow &window,
+                                             int block)
+{
+    int id_x = hipBlockIdx_x * hipBlockDim_x + hipThreadIdx_x;
+
+    int outBlock = id_x * block;
+    if (outBlock >= outEnd)
+        return;
+
+    int blockEnd = std::min(outBlock + block, outEnd);
+    double inBlockRaw = outBlock * scale;
+    int inBlockRounded = static_cast<int>(inBlockRaw);
+    float inPos = inBlockRaw - inBlockRounded;
+    float fscale = scale;
+    float tempBuf[RPPT_MAX_CHANNELS] = {0.0f}; // Considering max channels as 3
```

Comment:
@snehaa8 Again tempBuf[3] has been changed to tempBuf[RPPT_MAX_CHANNELS] due to https://github.com/r-abishek/rpp/pull/267#discussion_r1580940354 but the comment still says 3.
L54 above also says 3